src/code-templates/services/order-service/entry-points/api/server.ts

```diff
@@ -60,20 +61,24 @@ async function openConnection(
 function handleRouteErrors(expressApp: express.Application) {
   expressApp.use(
     async (
-      error: any,
+      error: unknown,
       req: express.Request,
       res: express.Response,
+      // Express requires next function in default error handlers
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
       next: express.NextFunction
     ) => {
-      if (typeof error === 'object') {
-        if (error.isTrusted === undefined || error.isTrusted === null) {
+      if (error instanceof AppError) {
+        if (error?.isTrusted === undefined || error.isTrusted === null) {
           error.isTrusted = true; // Error during a specific request is usually not fatal and should not lead to process exit
         }
+
+        res.status(error.HTTPStatus);
       }
       // ✅ Best Practice: Pass all error to a centralized error handler so they get treated equally
-      await errorHandler.handleError(error);
+      errorHandler.handleError(error);
 
-      res.status(error?.HTTPStatus || 500).end();
```

Comment:
⚠️ I think it's needed (sending back the right status)?

Reply:
oopsy 😅 fixed

Reply:
Always told that testing can not be trusted